### PR TITLE
add area and volume calculations

### PR DESCRIPTION
**Pull request by BeastyBlacksmith**
Implement calculation of area and volume
This is one of my first PRs so I'm not sure, if it meets the specifics of this package entirely.
The `area` function may or may not be a substitute for https://github.com/JuliaGeometry/GeometryTypes.jl/blob/649fe6f74de19a29c50f8983034e465f22184136/src/polygons.jl#L16, which is either wrong or I didn't fully understand its purpose.

### CODE REVIEW

**Comment by CarpeNecopinum** [comment]
It's not a replacement for the area method of `polygons.jl` (which e.g. handles non-convex polygons without having to triangulate them first). But what you're suggesting looks like a valuable addition to me.

Adding that gts parser and a relatively large mesh with it for testing is a bit overkill though, imho. I would suggest to just build a small mesh (e.g. a cube, has the advantage that it's obvious how large the volume/area should be) directly in the code and use that for testing.

**Comment by SimonDanisch** [comment]
Sorry for completely ignoring this! @CarpeNecopinum probably has a better idea ;) I basically just copy & pasted the area code for a single purpose usage -.-

**Comment by BeastyBlacksmith** [comment]
how do i get a properly oriented cube?
I tried
```julia
    cube = HyperRectangle(Vec3f0(-0.5), Vec3f0(1))
    cube_faces = decompose(Face{3,Int32}, cube)
    cube_vertices = decompose(Point{3,Float32}, cube) |> Array
```
but the volume computes to zero...

**Comment by CarpeNecopinum** [comment]
When you rebase against master, it makes new commits that have the same messages, but different hashes. You should have merged master instead.

Happened to me earlier today, too. 😅

**Comment by BeastyBlacksmith** [comment]
that indeed does look better

**Comment by CarpeNecopinum** [comment]
I just found out that the faces of a 3d HyperRectangle are inconsistently oriented, so you getting a volume of zero is actually the "correct" result (i.e. your algorithm does nothing wrong here).

You should be able to test your algorithm with the branch in #183, there I fixed the orientation so all faces point outwards (with counter-clockwise winding order).

**Comment by BeastyBlacksmith** [comment]
Thanks! This is fine now

**Comment by SimonDanisch** [comment]
Is this mergable then?

File: src/algorithms.jl [hunk 1]
```diff
@@ -1,3 +1,9 @@
+function normal( v1, v2, v3 )
+        a = v2 - v1
+        b = v3 - v1
+        cross(a, b)
+end
+
```
Comment (CarpeNecopinum) [inline src/algorithms.jl]: Wouldn't call this `normal`, as the actual normal is the normalized version of this.

Comment (BeastyBlacksmith) [reply src/algorithms.jl]: too much of a pun, if I name it `unnormal`?

Comment (CarpeNecopinum) [reply src/algorithms.jl]: `edge_cross_product` maybe? Personally I'd probably just keep it as a local function inside the other ones. Even though you'd have the same code twice then...

Comment (CarpeNecopinum) [reply src/algorithms.jl]: Re: Silly names. What about `normal_times_twice_the_area`? 🤓

Comment (SimonDanisch) [reply src/algorithms.jl]: maybe something like `orthogonal_vector`?

**Comment by BeastyBlacksmith** [comment]
`orthogonal_vector` then. This is ready to merge.

**Comment by SimonDanisch** [comment]
Thanks!

**Pull request merged**